substrate/client/rpc-spec-v2/src/chain_head/chain_head.rs

```diff
@@ -432,9 +433,14 @@ where
 	fn chain_head_unstable_unpin(
 		&self,
 		follow_subscription: String,
-		hash: Block::Hash,
+		hash_or_hashes: ListOrValue<Block::Hash>,
 	) -> RpcResult<()> {
-		match self.subscriptions.unpin_block(&follow_subscription, hash) {
+		let hashes = match hash_or_hashes {
+			ListOrValue::Value(hash) => vec![hash],
```

Comment:
perhaps you could use the iterator impl instead and just do std::iter::once(hash) instead of creating a vec here

Reply:
That makes total sense, have made the lower levels of `unpin_blocks` take `impl IntoIterator< ...>`. 

To remove the allocation, I've called into `self.subscriptions.unpin_blocks` with different iterators. Since, on one hand, we have `std::iter::Once` for a single value, and on the other, we have `vec`. Using some type-erasure would still make an allocation :D